weave/integrations/openai/openai_sdk.py

```diff
@@ -303,6 +304,14 @@ def should_use_accumulator(inputs: dict) -> bool:
 def openai_on_input_handler(
     func: Op, args: tuple, kwargs: dict
 ) -> ProcessedInputs | None:
+    attrs = call_context.call_attributes.get()
+
+    completion = args[0]
+    if hasattr(completion, "_client") and hasattr(completion._client, "_base_url"):
+        base_url = str(completion._client._base_url)
+        attrs["api_endpoint"] = base_url
```

Comment:
`api_endpoint` as a top-level attribute needs to be carefully considered.
1. Once you add this here, there is going to be an expectation to match this elsewhere. It is going to end up having queries against it etc... this is almost as important as thinking about a DB field. 
2. We would typically put weave-specific values under the `weave` sub-key. However in this case, it is closer to user-defined so it seems OK to put here.
3. Do we want to call it `client_base_url`? Why change the name from the source?